src/threading.c

```diff
@@ -537,11 +537,15 @@ JL_DLLEXPORT int jl_in_threaded_region(void)
 
 JL_DLLEXPORT void jl_enter_threaded_region(void)
 {
+    if (jl_atomic_load_relaxed(&jl_current_task->tid) != 0)
+        return;
```

Comment:
This seems to require https://github.com/JuliaLang/julia/pull/43420#issuecomment-995130391

Reply:
Yeah, this commit needs to be dropped. I need to instead actually make sure every thread can safely change this.

Reply:
okay, I have tried to fix this now